source/Octopus.Tentacle/Scripts/ScriptStateStoreFactory.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using Octopus.Tentacle.Contracts;
+using Octopus.Tentacle.Util;
+
+namespace Octopus.Tentacle.Scripts
+{
+    internal class ScriptStateStoreFactory : IScriptStateStoreFactory
```

Comment:
It looks like this custom code does what Autofac delegate factory does: https://autofac.readthedocs.io/en/latest/advanced/delegate-factories.html

Reply:
Would this require the consumer to take a dependency on `IOctopusFileSystem` so it could pass it to the delegate factory?

Reply:
No, that's the beauty of this solution. Autofac will inject what it can. E.g. HubClient takes 11 dependencies and but the client needs to provide only 3 of them: https://github.com/OctopusDeploy/CloudPortal/blob/7d905aac2bb4826a89d879a5d19f0cd29ec3c950/source/CloudPortal.Core/Hub/HubClient.cs#L38

Reply:
Ok I'll revisit this. 👍